src/utils.ts

```diff
@@ -58,6 +65,24 @@ export function useSearch(initialMovies: IMockMovie[]) {
     setSearchQuery,
     setFilterParam,
     searchedMovies,
-    searchMovies
+    searchMovies,
   };
 }
+
+export function useMovies() {
+  const movies: Ref<IMockMovie[]> = ref([]);
+  const error: Ref<string | null> = ref(null);
+
+  const getAllMovies = async () => {
+    try {
+      const endpoint = "/movies.json";
+      const response = await apiClient(endpoint);
+      movies.value = response.movies;
```

Comment:
This is deffenetlly won't work for you if your request fails. Look closer to Promise chain you have created

Reply:
fixed